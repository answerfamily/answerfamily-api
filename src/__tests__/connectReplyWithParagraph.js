jest.mock('../dataloaders/auth0UserLoaderFactory');

const {
  loadESFixtures,
  unloadESFixtures,
  loadMongoFixtures,
  unloadMongoFixtures,
} = require('../../test/fixtures');
const gql = require('../../test/gql');
const {
  ES_FIXTURES,
  MONGO_FIXTURES,
} = require('../__fixtures__/articleParagraphReply');
const mongoClient = require('../lib/mongoClient');

describe('connectReplyWithParagraph', () => {
  beforeAll(async () => {
    await loadESFixtures(ES_FIXTURES);
    await loadMongoFixtures(MONGO_FIXTURES);
  });

  afterAll(async () => {
    await unloadESFixtures(ES_FIXTURES);
    await unloadMongoFixtures(MONGO_FIXTURES);
  });

  it('connects reply and paragraph', async () => {
    const ids = { replyId: 'r1', paragraphId: 'p2' };

    const { errors, data } = await gql`
      mutation($replyId: String!, $paragraphId: String!) {
        connectReplyWithParagraph(
          replyId: $replyId
          paragraphId: $paragraphId
        ) {
          paragraphReplies {
            reply {
              id
            }
            user {
              id
              name
            }
          }
        }
      }
    `(ids, {
      userPromise: Promise.resolve({ iss: 'user-id' }),
    });

    expect(errors).toBeUndefined();
    expect(data.connectReplyWithParagraph).toMatchSnapshot();

    // cleanup
    const { db } = await mongoClient;
    await db
      .collection('paragraphReplies')
      .deleteOne({ replyId: 'r1', paragraphId: 'p2' });
  });

  // it('cannot connect already connected reply and paragraph', async () => {});

  // it('blocks non-logged in users', async () => {});
});
