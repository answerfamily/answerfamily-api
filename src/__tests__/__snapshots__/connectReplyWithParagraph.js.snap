// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connectReplyWithParagraph connects reply and paragraph 1`] = `
Object {
  "paragraphReplies": Array [
    Object {
      "reply": Object {
        "id": "r2",
      },
      "user": Object {
        "id": "author1",
        "name": "[mock] name of author1",
      },
    },
    Object {
      "reply": Object {
        "id": "r1",
      },
      "user": Object {
        "id": "user-id",
        "name": "[mock] name of user-id",
      },
    },
  ],
}
`;
